scripts/locations_data_load/locations_lambda.py

```diff
@@ -0,0 +1,283 @@
+        if locations_item.get("managingOrganization") == "":
+            locations_lookup_field_value = locations_item.get("lookup_field")
+
+            if locations_lookup_field_value:
+                for org_item in org_items:
+                    org_identifier_value = org_item.get("identifier", {}).get(
+                        "value", ""
+                    )
+                    if org_identifier_value == locations_lookup_field_value:
+                        org_id = org_item.get("id")
+                        locations_table.update_item(
+                            Key={"id": locations_id},
+                            UpdateExpression="SET managingOrganization = :val",
+                            ExpressionAttributeValues={":val": org_id},
+                        )
+
+
+odscode_file_path = "./ODS_Codes.xlsx"
+
+
+def read_excel_values(odscode_file_path):
+    # Read values from the Excel file
+    excel_data = pd.read_excel(odscode_file_path)
+    param1_values = excel_data["ODS_Codes"].tolist()
+
+    # # Hardcoded values for param2
+    param2_value = [
+        "OrganizationAffiliation:primary-organization",
+        "OrganizationAffiliation:participating-organization",
+    ]
+
+    # list of dictionaries with the desired format
+    params = [
+        {"primary-organization": val, "_include": param2_value} for val in param1_values
+    ]
+
+    return params
+
+
+token_api_endpoint = "https://beta.ods.dc4h.link//authorisation/auth/realms/terminology/protocol/openid-connect/token"
+
+
+# DynamoDB table name
+dynamodb_table_name = "locations"
+
+
+# def write_to_json(output_file_path, processed_data):
+#     import json
+#     with open(output_file_path, "a") as output_file:
+#         json.dump(processed_data, output_file, indent=2)
+#         output_file.write("\n")
+
+
+# # Iterate over Excel values and make API requests
+def fetch_organizations():
+    api_endpoint = "https://beta.ods.dc4h.link/fhir/OrganizationAffiliation?active=true"
+    failed_to_fetch = "Failed to fetch data from the ODS API."
+    odscode_params = read_excel_values(odscode_file_path)
+    headers = get_headers()
+    for odscode_param in odscode_params:
+        # Call the function to read from the ODS API and write to the output file
+        response_data = read_ods_api(api_endpoint, headers, params=odscode_param)
+
+        # Process and load data to json file
+        if response_data:
+            organizations = response_data.get("entry", [])
+            processed_data = process_organizations(organizations)
+            write_to_dynamodb(dynamodb_table_name, processed_data)
+            # output_file_path = "./location.json"
+            # write_to_json(output_file_path, processed_data)
+
+        else:
+            print(failed_to_fetch)
+
+    if response_data:
+        print("Data fetched successfully.")
+
+    else:
+        print(failed_to_fetch)
+
+
+# fetch Y code organizations
+def fetch_Y_organizations():
```

Comment:
Could be wrong but I don't think this is covered by unit tests which means we are not testing teh logic ? Could we ?